pyo3-build-config/src/impl_.rs

```diff
@@ -1118,6 +1112,7 @@ pub fn cross_compiling_from_to(
 ///
 /// This must be called from PyO3's build script, because it relies on environment
 /// variables such as `CARGO_CFG_TARGET_OS` which aren't available at any other time.
+#[allow(dead_code)]
```

Comment:
Not actionable now, but after #5486 we can make use of `#[expect(...)]` for these.